### ISSUE TITLE
Underutilized GPU

### ISSUE THREAD

**Issue by GEGR667** [cocktailpeanut/fluxgym#160]
Hi - More of a question about GPU utilization than an issue per se. (Congrats on the GUI, by the way - way more user friendly!!) 
After some initial teething problems, I've gotten Fluxgym to run without stalling, but I wonder if its underutilizing the GPU?
I'm using a 12gb 3060 with a 12th-gen i5 on 64gb of system RAM, and training the model using Flux.1 Dev. I'm under no illusion of speed considering its a 3060. I'm training a Lora with 14 images cropped to 512x512, and default Loragym settings, with the 12gb VRAM option.
On Windows Task Manager, it says the CPU is being used up to ~45%, while the GPU 1 NVIDIA GeForce RTX 3050 (using an e-gpu on Thunderbolt 4) is being used between 1% to 8%, with GPU VRAM usage between 1g/t to 7.0gb <<< seems a bit underutilized, no?
I haven't screwed around with using FP8, a smaller model (ie. Flux Schnell, or NF4) or other advanced settings.
So speed is what it is (its a 3060 12gb VRAM using Flux.1 Dev, after all), but are the GPU and VRAM being underutilized, and is there any way or setting to utilize more more of the GPU and VRAM processing power to speed things up?
Sorry if there's a better place to post these kinds of questions. I saw some posts about the initial teething problems I was having, so I figured the fluxgym GIT issues might be a decent place to ask.

**Comment by GEGR667** [comment]
Here's an update for those interested.
I restarted the process, because I put too few sample image every N steps, which I figured were taking up more time that it was worth. I also set reduced the training images to just a face at 512x512 from a face and pose at 512x512, so likely fewer parameters for the LoRa to learn. 17 images being trained on now from 14 before, so more images, but fewer sample images to interrupt processing time with (again, this is a 3060 12gb). 
CPU and GPU / GPU memory usage is the same as before. I've read something about not using UNET to reduce time, though it's counterintuitive (to me) since that's the model its training the LoRa from. I'm still curious is there's another setting to better optimize processing speed with. I could have also used a smaller model (ie. NF4), which might have sped things up. Thoughts?
After about 9 hours, I'm halfway through the training, with two LoRa checkpoints added (8 out of 16 epochs - I believe).
Also, I noticed that the Fluxgym WebGUI stops updated after a certain point. No more sample images or logs are produced on the WebGUI, though the training run continues to post sample images to the file directory, so that's how I know the process hasn't stalled.

**Comment by kitamo1981** [comment]
I have already trained several dozen Loras on Flux. On average I don't exceed 2200 steps for character and it take something about 7hours.

**Comment by GEGR667** [comment]
Thanks for your comment.
I've played around some more with FluxGym, and found that by increasing the Training Batch Size to 3 and the learning rate by 3x, you can cut down the training time using a 12gb 3060 for 16 512x512 images over 2400 steps from over about 11 hours, to just under 5 hours with little to no average loss or quality. Increasing the Training Batch Size makes considerably more use of the GPU's available memory.
The only problem is that you can't change the batch size in the web UI; you have to change the Training Batch Size in the app.py file. Open it, search "batch_size =" and change the value from 1 to 3. You can change the learning rate in the web UI.
I've also tried some of the cloud based implementations of Fluxgym, where you can rent a high-spec GPU. They seem to run fine, but there you can't change the training batch size.

**Comment by kitamo1981** [comment]
hmmm batch_size=3 give me an error. Need to check this. Can U paste here UR script settings?

**Comment by GEGR667** [comment]
Just guessing here: if you copy/pasted your script, you may need spaces between the = and 3, ie:  "batch_size = 3"
Here are four lines of the script I copy/pasted from my app.py. I only changed the 1 to a 3:
[[datasets]]
resolution = {resolution}
batch_size = 3
keep_tokens = 1
